Tools/ci-run.sh

```diff
@@ -3,125 +3,164 @@
 GCC_VERSION=${GCC_VERSION:=8}
 
 # Set up compilers
-if [ "${OS_NAME##ubuntu*}" == "" -a "$TEST_CODE_STYLE" != "1" ]; then
+if [[ $TEST_CODE_STYLE == "1" ]]; then
+  echo "Skipping compiler setup"
+elif [[ $OSTYPE == "linux-gnu"* ]]; then
+  echo "Setting up linux compiler"
   echo "Installing requirements [apt]"
   sudo apt-add-repository -y "ppa:ubuntu-toolchain-r/test"
   sudo apt update -y -q
   sudo apt install -y -q ccache gdb python-dbg python3-dbg gcc-$GCC_VERSION || exit 1
-  if [ -z "${BACKEND##*cpp*}" ]; then
+  if [[ $BACKEND == *"cpp"* ]]; then
     sudo apt install -y -q g++-$GCC_VERSION || exit 1
+    ALTERNATIVE_ARGS=" --slave /usr/bin/g++ g++ /usr/bin/g++-$GCC_VERSION"
```

Comment:
We should make sure that `ALTERNATIVE_ARGS` isn't just taken from somewhere else by (re-)setting it in all cases.

Reply:
I do not understand where else `ALTERNATIVE ARGS` can come from?

Reply:
Someone could decide to add such a variable to the Github build environment.

Besides, I don't consider it good style to rely on uninitialised variables.

Reply:
[resolved](https://github.com/cython/cython/pull/4324/commits/5eff4150cd36f24197f665740d060467b8034967)